src/Streamly/Internal/Data/Stream/Zip.hs

```diff
@@ -8,6 +8,10 @@
 {-# LANGUAGE TypeFamilies              #-}
 {-# LANGUAGE UndecidableInstances      #-} -- XXX
 
+#if MIN_VERSION_base(4,17,0)
```

Comment:
Why is this required?

Reply:
```
$ cabal build all --enable-tests --enable-benchmarks --with-compiler=ghc-9.4.5 \
  --ghc-options="-w -Wtype-equality-requires-operators"
Build profile: -w ghc-9.4.5 -O1
...
[37 of 68] Compiling Streamly.Internal.Data.Stream.Zip

src/Streamly/Internal/Data/Stream/Zip.hs:185:2923: warning: [-Wtype-equality-requires-operators]
    The use of ‘~’ without TypeOperators
    will become an error in a future GHC release.
    Suggested fix: Perhaps you intended to use TypeOperators
    |
185 | LIST_INSTANCES(ZipSerialM)
    |
 ```

Reply:
I removed the `CPP` guard but left the `-XTypeOperators` extension.